### PR TITLE
feat: Add resource controlled vocabulary

### LINKED ISSUE #164: chore: Populate resources samples from external source
To avoid freetext of resource names, let's populate resource ahead of time or use a good validator for vocabularies. See https://github.com/uw-ssec/post-disaster-comms/issues/42 related to this.

### PR DESCRIPTION
**Pull request by lsetiawan**
This pull request introduces several changes to the `support_sphere` project, primarily focusing on adding new models and updating existing ones to enhance the resource management system. The most important changes include the addition of the `ResourceCV` model, modifications to the `Resource` and `ResourceType` models, and updates to the database sample data script to populate new resource types and controlled vocabulary.

### New Models and Relationships:

* [`src/support_sphere_py/src/support_sphere/models/public/resource_cv.py`](diffhunk://#diff-19802de6232ebc3a97a5bb6d811d4a9c43041f36b77995baf3ddeae4075f26f9R1-R28): Added a new `ResourceCV` model to represent resource controlled vocabulary items. (`[src/support_sphere_py/src/support_sphere/models/public/resource_cv.pyR1-R28](diffhunk://#diff-19802de6232ebc3a97a5bb6d811d4a9c43041f36b77995baf3ddeae4075f26f9R1-R28)`)
* [`src/support_sphere_py/src/support_sphere/models/public/resource.py`](diffhunk://#diff-0a902dfe2186914754b8a0bc1f13c45e45c31b5e1003fd9f78d6ca7aa843cc3fL39-R48): Updated the `Resource` model to include a relationship with `ResourceCV` and replaced the `name` and `description` fields with `resource_cv_id`. (`[src/support_sphere_py/src/support_sphere/models/public/resource.pyL39-R48](diffhunk://#diff-0a902dfe2186914754b8a0bc1f13c45e45c31b5e1003fd9f78d6ca7aa843cc3fL39-R48)`)

### Model Updates:

* [`src/support_sphere_py/src/support_sphere/models/public/__init__.py`](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R15): Imported the new `ResourceCV` model and added it to the `__all__` list. (`[[1]](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R15)`, `[[2]](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R43)`)
* [`src/support_sphere_py/src/support_sphere/models/public/resource_type.py`](diffhunk://#diff-5b7b32b83944726536b84d2cfabfd71820b33ff85fd5957f8eb0ab8daa52d2ffR17-R18): Added an optional `description` field to the `ResourceType` model. (`[[1]](diffhunk://#diff-5b7b32b83944726536b84d2cfabfd71820b33ff85fd5957f8eb0ab8daa52d2ffR17-R18)`, `[[2]](diffhunk://#diff-5b7b32b83944726536b84d2cfabfd71820b33ff85fd5957f8eb0ab8daa52d2ffR30)`)

### Database Sample Data:

* [`src/support_sphere_py/tests/resources/data/resources_cv.csv`](diffhunk://#diff-a6adbf3ac0808dc772861567fa591419ea93fa9aa55abba1777130d7e4c8662eR1-R92): Added a CSV file containing resource controlled vocabulary items and their descriptions. (`[src/support_sphere_py/tests/resources/data/resources_cv.csvR1-R92](diffhunk://#diff-a6adbf3ac0808dc772861567fa591419ea93fa9aa55abba1777130d7e4c8662eR1-R92)`)
* [`src/support_sphere_py/tests/resources/scripts/update_db_sample_data.py`](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bL8-R8): Updated the script to populate the new `ResourceType` and `ResourceCV` data into the database. (`[[1]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bL8-R8)`, `[[2]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bR21-R49)`, `[[3]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bR177-R179)`)

### Configuration Updates:

* [`pixi.toml`](diffhunk://#diff-b092682f0ad15a352f9fa2b0a67c992454b586275ee76b74a5839ab9afdf9894R79): Added a new dependency `setup-and-load` which depends on `setup-infra` and `setup-db-data-via-k8s-job`. (`[pixi.tomlR79](diffhunk://#diff-b092682f0ad15a352f9fa2b0a67c992454b586275ee76b74a5839ab9afdf9894R79)`)This pull request introduces several changes to the `support_sphere` project, focusing on adding a new `ResourceCV` model, updating dependencies, and enhancing database sample data population scripts. The most important changes include adding the `ResourceCV` model, updating the `Resource` model to use `ResourceCV`, and populating the database with resource types and controlled vocabulary items.

### New Model Addition:
* [`src/support_sphere_py/src/support_sphere/models/public/resource_cv.py`](diffhunk://#diff-19802de6232ebc3a97a5bb6d811d4a9c43041f36b77995baf3ddeae4075f26f9R1-R28): Added the `ResourceCV` model representing a list of resource controlled vocabulary items. (`[src/support_sphere_py/src/support_sphere/models/public/resource_cv.pyR1-R28](diffhunk://#diff-19802de6232ebc3a97a5bb6d811d4a9c43041f36b77995baf3ddeae4075f26f9R1-R28)`)

### Model Updates:
* [`src/support_sphere_py/src/support_sphere/models/public/resource.py`](diffhunk://#diff-0a902dfe2186914754b8a0bc1f13c45e45c31b5e1003fd9f78d6ca7aa843cc3fL39-R48): Updated the `Resource` model to include a foreign key to `ResourceCV` and established a relationship between `Resource` and `ResourceCV`. (`[src/support_sphere_py/src/support_sphere/models/public/resource.pyL39-R48](diffhunk://#diff-0a902dfe2186914754b8a0bc1f13c45e45c31b5e1003fd9f78d6ca7aa843cc3fL39-R48)`)
* [`src/support_sphere_py/src/support_sphere/models/public/__init__.py`](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R15): Imported `ResourceCV` and added it to the `__all__` list. (`[[1]](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R15)`, `[[2]](diffhunk://#diff-2fd7615f1a2aa0dcf722cc23576a73dfd71ba3f0009a670526f6303ab10e8ef6R43)`)

### Database Sample Data Population:
* [`src/support_sphere_py/tests/resources/data/resources_cv.csv`](diffhunk://#diff-a6adbf3ac0808dc772861567fa591419ea93fa9aa55abba1777130d7e4c8662eR1-R92): Added a CSV file containing sample data for `ResourceCV` items. (`[src/support_sphere_py/tests/resources/data/resources_cv.csvR1-R92](diffhunk://#diff-a6adbf3ac0808dc772861567fa591419ea93fa9aa55abba1777130d7e4c8662eR1-R92)`)
* [`src/support_sphere_py/tests/resources/scripts/update_db_sample_data.py`](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bL8-R8): Added functions to populate `ResourceType` and `ResourceCV` data into the database and updated the main execution block to call these functions. (`[[1]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bL8-R8)`, `[[2]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bR21-R49)`, `[[3]](diffhunk://#diff-7ca4dd666d4bf5c1344d9b43a9354dea92bc700f7f655cbd3865e752d9f2b72bR177-R179)`)

### Configuration Updates:
* [`pixi.toml`](diffhunk://#diff-b092682f0ad15a352f9fa2b0a67c992454b586275ee76b74a5839ab9afdf9894R79): Added a new dependency `setup-and-load` that depends on `setup-infra` and `setup-db-data-via-k8s-job`. (`[pixi.tomlR79](diffhunk://#diff-b092682f0ad15a352f9fa2b0a67c992454b586275ee76b74a5839ab9afdf9894R79)`)

## Related Issue

Resolved #164

### CODE REVIEW

File: src/support_sphere_py/tests/resources/data/resources_cv.csv
Comment (nikiburggraf) [inline src/support_sphere_py/tests/resources/data/resources_cv.csv:1]: Should this file also define what Type (skill, consumable, durable, etc) each of these resources is so we can establish that relationships when we get the database started up? Might be some extra work, but it's probably gotta get done at some point, and I wonder if this is the right place to get that set up

**Review by nikiburggraf — COMMENTED** [review]
Looks good for the most part, just a question on if the default resources we're gonna load should be associated with types on db startup

File: src/support_sphere_py/tests/resources/data/resources_cv.csv
Comment (lsetiawan) [reply src/support_sphere_py/tests/resources/data/resources_cv.csv:1]: Yea, I was thinking about that. I do have a table of that. I suppose for an initial development, we can do that. I'll set that up and then on a separate PR push the code for it.

Comment (nikiburggraf) [reply src/support_sphere_py/tests/resources/data/resources_cv.csv:1]: Alright! In that case, I can approve this PR then :)

**Pull request merged**